### PR TITLE
Fixes #24033 - Refactor the subscriptions-page

### PR DESCRIPTION
**Pull request by sharvit**
**This PR contains 2 other commits:**
https://github.com/Katello/katello/pull/7524
https://github.com/Katello/katello/pull/7508

**Context**
* Move the SubscriptionsPage.js state to redux
* Rename scenes/Tasks to scenes/TasksMonitor and refactor it
    * Allow multiple monitoring
    * Allow to cancel monitor by id
    * Allow to manually run the monitor action
* Instead of monitor tasks by search and then monitor the specific task, I changed it to only monitor tasks by search and filter them to find the one `inProgress`.

**Disclaimer**
I believe using `redux-reselect` will boost performance but I couldn't use it since it still hasn't came into foreman.
Migrate it later should be an easy job.

**TODO**
- [x] Testing

### CODE REVIEW

File: app/controllers/katello/react_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module Katello
   class ReactController < ::ApplicationController
+    helper ::Rails.application.routes.url_helpers
```
Comment (sharvit) [inline app/controllers/katello/react_controller.rb]: Not related

**Comment by sharvit** [comment]
@waldenraines @amirfefer @tstrachota - Can you review please?

**Comment by sharvit** [comment]
@johnpmitsch @waldenraines - I afraid it is not as simple as it looks.

This PR break files and methods into a different structure.
Since many bugs got fixes and new features added we have the risk of bringing old bugs to life and breaking the new features.

I afraid it means we can't rebase and we need to rewrite 70% of the refactoring.

I will give it another look today to see if the rebase is actually possible.

**Comment by waldenraines** [comment]
@sharvit is this now being broken up into smaller PRs?

https://github.com/Katello/katello/pull/7775

**Comment by sharvit** [comment]
Yes @waldenraines 👍 

Because it is difficult to rebase, I'm trying to get small parts into smaller PRs.

**Comment by waldenraines** [comment]
> Because it is difficult to rebase, I'm trying to get small parts into smaller PRs.

Should we close this then?

**Comment by sharvit** [comment]
Yes!

**Pull request closed without merging**